apiserver/facades/client/client/status.go

```diff
@@ -1190,8 +1190,10 @@ func (c *statusContext) makeMachineStatus(
 	constraints := c.allConstraints.Machine(machineID)
 	status.Constraints = constraints.String()
 
-	hc := c.allInstances.HardwareCharacteristics(machineID)
-	if hc != nil {
+	hc, err := machineService.HardwareCharacteristics(ctx, machineUUID)
+	if err != nil {
+		logger.Debugf("error fetching hardware characteristics: %v", err)
```

Comment:
So, I had seen this one and I had purposely left it as a generic error because it's just a debug log on status. I added the check now so we have 2 different debug lines anyways.